### PR TITLE
Several Fixes (freerdp3 command line, able to be run on multiple distros, fixing a typo)

### PR DESCRIPTION
**Pull request by GreatNovaDragon**
… to install all officially supported apps in one go, also fix a typo

This PR contains Several Fixes.
With FreeRDP3, the command line interface has been reworked, making the old one non-compatible.

Several of the commands given here made the baseline assumption that it is run on Ubuntu, like the assumption the command is always named "xfreerdp", when at least under Arch it is xfreerdp3. So the check for that is also in, and the option to supply an own command there. 
Or the base assumption that bc is installed (it isnt under arch)

It also adds an convenience argument to setup all officially supported apps at once. 

Also, there was an typo for the Windows launcher, telling us it belonged to "Micorosoft" instead of Microsoft

### CODE REVIEW

**Comment by GreatNovaDragon** [comment]
Should fix #81 
Might fix #79 
Affects #69 in that you would be able to just select the commands for sdl-freerdp3 and wlfreerdp3 (even if their support for RemoteApp is not existent right now)

**Comment by oskardotglobal** [comment]
Hello, thank you for your contribution.
So far this looks good, but I will have to test it on my machine

**Comment by oskardotglobal** [comment]
@LDprg Thoughts?

**Comment by LDprg** [comment]
Seems very good to me, a lot of great refinements. Thx for the contribution.

@oskardotglobal have you tested it?

However we also should make a proper review. (I will do one propably by the end of this week, if nobody does before)

**Comment by LDprg** [comment]
@oskardotglobal What is about our licence issue? Are we even permitted to merge this pr.

**Comment by oskardotglobal** [comment]
> @oskardotglobal What is about our licence issue? Are we even permitted to merge this pr.

Well, we're technically not allowed to redistribute or modify winapps in any way, but given that fmstrat isn't going to do anything about it, we're fine. Also, as the PR's author has signed the CLA its changes fall under the GPLv3 license, so at least that part is all clear

File: bin/winapps [hunk 1]
```diff
@@ -34,52 +33,59 @@ if [ -f "${HOME}/.local/share/winapps/run" ]; then
...
 	fi
 else
 	touch "${HOME}/.local/share/winapps/run"
 fi
 
-if [ -z "$(which xfreerdp)" ]; then
-	echo "You need xfreerdp!"
-	echo "  sudo apt-get install -y freerdp2-x11"
-	exit
+if [ -z "${FREERDP_COMMAND}" ]; then
```
Comment (LDprg) [inline bin/winapps:43]: We still should check whether the freerdp command is available and print a message

File: installer.sh [hunk 2]
```diff
@@ -1,6 +1,12 @@
 #!/usr/bin/env bash
 
-DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+if ! command -v bc &> /dev/null
```
Comment (LDprg) [inline installer.sh:3]: Do we really need to check if bc is installed? We like use it only in [here](https://github.com/winapps-org/winapps/blob/152a92412badea5e366c918be1cebd142624e377/install/inquirer.sh#L142), so we might just search for an alternative implementation without bc.

**Review by LDprg — CHANGES_REQUESTED** [review]
Great work! This improves code quality a lot and has many small improvment!

Just two small changes and we are ready to merge. Thx.

File: README.md [hunk 3]
```diff
@@ -149,31 +170,40 @@ Adding applications with custom icons and mime types to the installer is easy. S
...
 ./installer.sh --user                # Configure applications for the current user
 ./installer.sh --system              # Configure applications for the entire system
 ./installer.sh --user --uninstall    # Remove all configured applications for the current user
 ./installer.sh --system --uninstall  # Remove all configured applications for the entire system
 ```
+./installer.sh --user --setupAllOfficiallySupportedApps    # Configures all officially supported applications for the current user
+./installer.sh --system --setupAllOfficiallySupportedApps  # Configures all officially supported applications for the entire system
+```
 
 ## Shout outs
```
Comment (LDprg) [inline README.md:203]: The formatting of this seems a bit messed up. (wrong code block)

**Review by LDprg — CHANGES_REQUESTED** [review]

File: bin/winapps (hunk 1 above)
Comment (GreatNovaDragon) [reply bin/winapps:43]: How this basic check suffices

File: installer.sh (hunk 2 above)
Comment (GreatNovaDragon) [reply installer.sh:3]: I wont dare to touch what i dont understand, someone else has to do that.

**Review by LDprg — APPROVED** [review]
Seems good to me.

@oskardotglobal @AkechiShiro Can you add a a quick review so we can merge this.

**Review by oskardotglobal — APPROVED** [review]
Looks good

**Review by AkechiShiro — APPROVED** [review]
Did a quick review, LGTM, would need to test this at a later time to check if it's all working properly

**Comment by LDprg** [comment]
@AkechiShiro have you tested this pr?

I would like to merge this as soon as possible (of course if tested properly).

**Comment by AkechiShiro** [comment]
I haven't found the time to do so sorry @LDprg

Would you be able to test it ?

**Comment by LDprg** [comment]
I am also quite busy @AkechiShiro, not sure when I can test this. @oskardotglobal maybe you could take a look?

I would suggest anyone testing first just reports their find here. The worst (or best ;-) ) that could happen is that we have multiple reports at the same time.

**Comment by AkechiShiro** [comment]
It's bit annoying to test we need to setup a W11 VM manually using KVM no ?

**Comment by AkechiShiro** [comment]
Maybe I'm doing something wrong but running the installer.sh script shows there might be syntax errors : 
image

Can you test the script please ? @GreatNovaDragon ?

**Comment by oskardotglobal** [comment]
@AkechiShiro you can always use quickemu to set up the VM
I will see if I can get this to work on my Fedora machine too

**Comment by AkechiShiro** [comment]
Ran into this issue : https://github.com/quickemu-project/quickemu/issues/686

**Comment by AkechiShiro** [comment]
My laptop is rebooting when I run a VM inside a VM wasn't expecting this one...

I'll need to test on another laptop running natively ArchLinux, I will be able to most likely in 2 weeks.

**Comment by LDprg** [comment]
I will start writing a docker guide at some point for the legacy winnapps (and code for the rewrite). Docker should work prevent most of the issue with quickemu (because it is specifically build for portability and windows).

**Comment by oskardotglobal** [comment]
Yeah, I'm also kinda fed up with quickemu
Using Docker will help here

**Comment by oskardotglobal** [comment]
Alright, I'm gonna test it right now

**Comment by oskardotglobal** [comment]
Ok, so I've caught a syntax error which caused me to not be able to run the script, but I've fixed that

**Comment by oskardotglobal** [comment]
Unless anyone else has anything to say, I say this is ready to merge

**Comment by AkechiShiro** [comment]
Thanks a lot @oskardotglobal and sorry for not having been able to wholefully test it

**Pull request merged**